lib/nokogiri/css/parser.rb

```diff
@@ -8,9 +8,15 @@
 
 
 require 'nokogiri/css/parser_extras'
+
 module Nokogiri
   module CSS
     class Parser < Racc::Parser
+
+
+def unescape_css_identifier(identifier)
+  identifier.gsub(/\\(?:([^0-9a-fA-F])|([0-9a-fA-F]{1,6})\s?)/){ |m| $1 || [$2.hex].pack('U') }
```

Comment:
it matches a  \ followed by any character other than hex characters, so  \: , etc  - or any 1-6 digit hex number potentially followed by whitespace - these are the two methods of escaping characters in css ids, classes, etc